python/deltalake/writer.py

```diff
@@ -132,8 +132,28 @@ def write_deltalake(
     :param overwrite_schema: If True, allows updating the schema of the table.
     :param storage_options: options passed to the native delta filesystem. Unused if 'filesystem' is defined.
     """
+
     if _has_pandas and isinstance(data, pd.DataFrame):
-        data = pa.Table.from_pandas(data)
+        if schema is not None:
+            data = pa.Table.from_panda(data, schema=schema)
+        else:
+            _data = pa.Table.from_pandas(data)
+            _schema = _data.schema
+            schema_out = []
+            for _field in _schema:
```

Comment:
Would you be willing to pull this out into a helper function (`_delta_arrow_schema_from_pandas`)?

Reply:
Really like this idea.  Factored this out into a helper function, but I'd prefer it to be public rather than private.  Would like to leverage this helper function elsewhere.  Thoughts?

Reply:
Sure, but if you make it public, then I'd suggest putting it in `schema.py`.

Reply:
Done.